mnemonic.c

```diff
@@ -450,10 +450,12 @@ mn_decode (char *src, void *dest, int destsize)
 
   while ((index = mn_next_word_index (&src)) != 0)
     {
-      if (index == 0 && *src != 0)
-	return MN_EWORD;
       (void) mn_decode_word_index (index, dest, destsize, &offset);
     }
-  (void) mn_decode_word_index (MN_EOF, dest, destsize, &offset);
+  if (index == 0 && *src != 0)
+    return MN_EWORD;
+  int status = mn_decode_word_index (MN_EOF, dest, destsize, &offset);
```

Comment:
C90?! _[Checks my watch]_ That standard's been obsolete for 15 years. Should I check it with a BCPL compiler too? :stuck_out_tongue_winking_eye: